### DISCUSSION TITLE
IndexError: list index out of range

### DISCUSSION THREAD

**Discussion by NyamdavaaAyush** [PabloLec/RecoverPy#263] [Q&A]
after the search, it found about 30 results, but there was no result list in left side and when i clicked buttons Open, the program breaks down with following error.

Any idea?

`/usr/local/lib/python3.8/dist-packages/recoverpy/ui/screens/screen_search.py:97 in on_button_pressed                                                      
│ ❱  97 │   │   │   │   │   self._grep_result_list.grep_results`

**Comment by PabloLec** [comment]
Hello,

It is indeed a bug. You're seeing this error because when you click on the "Open" button, the program can't find the selected item in the list.

This might be due to a bug in the UI library, which is still in beta. I'm behind on several versions, and these versions have introduced breaking changes.

Could you let me know your :
- OS: `lsb_release -a`
- Shell: `echo $SHELL`
- Terminal: `echo $TERM`

Regardless, this bug may be challenging to replicate. The best approach might be for me to simply update my TUI library. In the meantime, you can likely use version 1. The functionalities were the same, but the UI was a bit more basic.

`pip install "recoverpy>=1.0,<2.0"`

**Reply by phinix345** [reply]
Please help me IndexError: list index out of range my Avatarify won't change picture

**Comment by phinix345** [comment]
IndexError: list index out of range I get this error when I changed pictures in Avatarify

**Reply by PabloLec** [reply]
Hello,

The bug mentioned in this discussion should have been resolved since version 2.1.0. We can't rule out the possibility that the bug still exists or is present elsewhere. If that's the case, we will look into retrieving logs via the debug mode that was added in the meantime.

But before that, can you give me a bit more context:
- What exactly is the situation here? I searched for Avatarify on Google and I'm not sure I see the connection. Does the error occur while using RecoverPy?
- What is your system? Distribution and, if possible, terminal
- On what type of partition is the search being performed? NTFS, EXT4, etc.
- And finally, what is the version of RecoverPy being used?

**Comment by davina420** [comment]
![image](https://github.com/PabloLec/RecoverPy/assets/151196600/eb59aed6-e735-46ee-b3b2-0c13bb13bb13)
Please I’m stucked
